File: core/src/main/java/com/sequenceiq/cloudbreak/reactor/api/event/EventSelectorUtil.java
package com.sequenceiq.cloudbreak.reactor.api.event;

public class EventSelectorUtil {
    private EventSelectorUtil() {
    }

    public static String selector(Class clazz) {
        return clazz.getSimpleName().toUpperCase();
    }

    public static String failureSelector(Class clazz) {
        return clazz.getSimpleName().toUpperCase() + "_ERROR";
    }
}

File: core/src/main/java/com/sequenceiq/cloudbreak/reactor/api/event/EventSelectorUtil.kt
package com.sequenceiq.cloudbreak.reactor.api.event

object EventSelectorUtil {

    fun selector(clazz: Class<Any>): String {
        return clazz.simpleName.toUpperCase()
    }

    fun failureSelector(clazz: Class<Any>): String {
        return clazz.simpleName.toUpperCase() + "_ERROR"
    }
}
